components/server/src/user/token-service.ts

```diff
@@ -42,6 +55,54 @@ export class TokenService implements TokenProvider {
         return promise;
     }
 
+    // EXPERIMENT(sync_refresh_token_exchange)
+    private async doGetTokenForHostSync(userId: string, host: string): Promise<Token | undefined> {
+        const user = await this.userDB.findUserById(userId);
+        if (!user) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, `User (${userId}) not found.`);
+        }
+        const identity = this.getIdentityForHost(user, host);
+
+        const doRefreshToken = async () => {
+            // Check: Current token so we can actually refresh?
+            const token = await this.userDB.findTokenForIdentity(identity);
+            if (!token) {
+                return undefined;
+            }
+
+            const aboutToExpireTime = new Date();
+            aboutToExpireTime.setTime(aboutToExpireTime.getTime() + 5 * 60 * 1000);
```

Comment:
[nit] will a provider's token always expired in 5 minutes, then never make it works? 🤣